lib/message_bus.rb

```diff
@@ -503,6 +503,23 @@ def last_id(channel, site_id = nil)
     backend_instance.last_id(encode_channel_name(channel, site_id))
   end
 
+  # Get the ID of the last message published on multiple channels
+  #
+  # @param [Array<String>] channels - array of channels to fetch
+  # @param [String] site_id - the ID of the site by which to filter
+  #
+  # @return [Hash] the channel-specific IDs of the last message published to each requested channel
+  def last_ids(*channels, site_id: nil)
+    encoded_channel_names = channels.map { |c| encode_channel_name(c, site_id) }
+    ids = backend_instance.last_ids(*encoded_channel_names)
+
+    result = {}
```

Comment:
I'm curious, did you benchmark this change? Is it faster/lower on memory?

Reply:
I didn't... but I guess I should

```
require 'benchmark/ips'

Benchmark.ips do |x|
  channels = ["one", "two", "three", "four", "five"]
  ids = [1, 2, 3, 4, 5]

  x.report("zip") do 
    channels.zip(ids).to_h
  end

  x.report("each_with_index") do
    result = {}
    channels.each_with_index do |channel, index|
      result[channel] = ids[index]
    end
  end

  x.report("while") do
    result = {}
    i = 0
    while i < channels.size
      result[channels[i]] = ids[i]
      i += 1 
    end
  end

  x.report("transpose") do
    [channels, ids].transpose.to_h
  end

  x.compare!
end
```

```
Comparison:
               while:  1780741.1 i/s
           transpose:  1761829.2 i/s - same-ish: difference falls within error
                 zip:  1737068.4 i/s - same-ish: difference falls within error
     each_with_index:  1531362.7 i/s - 1.16x  (± 0.00) slower
```

😅 Looks like zip was faster after all. Will revert that commit

(ultimately this is negligible compared to the network round-trip for redis, so I'm not too worried)